### PR TITLE
[PLATFORM-997]: Spike: Migrate a veil repo to use GitHub Actions

### PR DESCRIPTION
**Pull request by MaeIsBad**
https://prima-assicurazioni-spa.myjetbrains.com/youtrack/issue/PLATFORM-997

### CODE REVIEW

File: veil-macros/src/fmt.rs [hunk 1]
```diff
@@ -6,7 +6,7 @@ use quote::ToTokens;
 use syn::spanned::Spanned;
 
 #[rustfmt::skip]
-/// Returns whether a syn::Type is an Option<T>
+/// Returns whether a syn::Type is an [`Option<T>`]
```
Comment (cottinisimone) [inline veil-macros/src/fmt.rs:9]:
```suggestion
/// Returns whether a [`syn::Type`] is an [`Option<T>`]
```

**Review by cottinisimone — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: Continuous Integration
+
+on: [push, pull_request]
```
Comment (cpiemontese) [inline .github/workflows/ci.yml:3]: Do we need the `pull_request` here? Isn't it implicit on push? :thinking:

Comment (MaeIsBad) [reply .github/workflows/ci.yml:3]: It's not, push only happens on branches that belong to this repo, but not on forks. If we want the CI to run for external contributions we need pull_request

Comment (cpiemontese) [reply .github/workflows/ci.yml:3]: Right! Noice

**Review by cpiemontese — DISMISSED** [review]
This seems ok, even though I'd be curious to test out a release

That said it seems we still need to remove the drone check

**Comment by MaeIsBad** [comment]
> This seems ok, even though I'd be curious to test out a release

as far as I can tell it seems to work(tested on a different branch with dry-run)

**Review by oliverbrowneprima — DISMISSED** [review]

**Comment by MaeIsBad** [comment]
I've made a small mistake in our umami PR, which causes our PRs to require the ugly "CI / Continuous Integration / test" check names. I propose we merge this as is and I'll fix this in a separate umami PR

**Review by cpiemontese — APPROVED** [review]

**Pull request merged**